flask_server.py

```diff
@@ -26,19 +27,49 @@ def get_settings():
         return json.load(settings_json)
 
 def render_base64_media(base64_media):
-    """
-    base64_media structure: 
-        0 - Data
-        1 - content type
-        2 - username
-    """
     rendered_base64_media = []
+    index_username = None
     for media in base64_media:
-        content_tag = "img" if "img" in media[1] else "video controls"
-        html_string = "<" + content_tag + " src=\"data:{};base64,".format(media[1]) + media[0] + "\"" + " class=\"rendered-stories\"></" + content_tag + ">"
-        rendered_base64_media.append(Markup(html_string))
+        base64_data, media_type, username = media
+        content_tag = "img" if "img" in media_type else "video controls"
+        if username != index_username:
+            rendered_base64_media.append(Markup(f"<hr><div class=\"username-text\">{username}</div><br>"))
+            rendered_base64_media.append(Markup(f"<{content_tag} src=\"data:{media_type};base64,{base64_data}\" class=\"rendered-stories\"></{content_tag}>"))
+            index_username = username
+        else:
+            rendered_base64_media.append(Markup(f"<{content_tag} src=\"data:{media_type};base64,{base64_data}\" class=\"rendered-stories\"></{content_tag}>"))
     return rendered_base64_media
 
+def get_gallery(basepath):
```

Comment:
Rename get_rendered_folders(path): 

Return list of rendered subfolder from path.

Reply:
Create new one just for media, get_rendered_media(path)